helix-core/src/main/java/org/apache/helix/controller/stages/ExternalViewComputeStage.java

```diff
@@ -78,6 +78,18 @@ public void process(ClusterEvent event) throws Exception {
         dataAccessor.getChildValuesMap(keyBuilder.externalViews());
 
     for (String resourceName : resourceMap.keySet()) {
+      IdealState idealState = cache._idealStateMap.get(resourceName);
+
+      // skip generate external view if the ExternalViewDisabled flag is set in IdealState
+      if (idealState != null && idealState.isExternalViewDisabled()) {
```

Comment:
What happens when the resource gets deleted? The isExternalViewDisabled in Idealstate is not available and we might write external views for all the resources.